lib/components/ArtistScreen/artist_screen_content_flexible_space_bar.dart

```diff
@@ -0,0 +1,198 @@
+import 'package:finamp/models/finamp_models.dart';
+import 'package:finamp/services/jellyfin_api.dart';
+import 'package:finamp/services/jellyfin_api_helper.dart';
+import 'package:finamp/services/queue_service.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:get_it/get_it.dart';
+
+import '../../models/jellyfin_models.dart';
+import '../../services/audio_service_helper.dart';
+import '../album_image.dart';
+import 'artist_item_info.dart';
+
+class ArtistScreenContentFlexibleSpaceBar extends StatelessWidget {
+  const ArtistScreenContentFlexibleSpaceBar({
+    Key? key,
+    required this.parentItem,
+    required this.isGenre,
+    required this.allSongs,
+    required this.albumCount,
+  }) : super(key: key);
+
+  final BaseItemDto parentItem;
+  final bool isGenre;
+  final Future<List<BaseItemDto>?> allSongs;
+  final int albumCount;
+
+  @override
+  Widget build(BuildContext context) {
+    GetIt.instance<AudioServiceHelper>();
+    QueueService queueService = GetIt.instance<QueueService>();
+
+    void playAllFromArtist(List<BaseItemDto> items) {
+      queueService.startPlayback(
+        items: items,
+        source: QueueItemSource(
+          type:
+              isGenre ? QueueItemSourceType.genre : QueueItemSourceType.artist,
+          name: QueueItemSourceName(
+              type: QueueItemSourceNameType.preTranslated,
+              pretranslatedName: parentItem.name ??
+                  AppLocalizations.of(context)!.placeholderSource),
+          id: parentItem.id,
+          item: parentItem,
+        ),
+        order: FinampPlaybackOrder.linear,
+      );
+    }
+
+    void shuffleAllFromArtist(List<BaseItemDto> items) {
+      queueService.startPlayback(
+        items: items,
+        source: QueueItemSource(
+          type:
+              isGenre ? QueueItemSourceType.genre : QueueItemSourceType.artist,
+          name: QueueItemSourceName(
+              type: QueueItemSourceNameType.preTranslated,
+              pretranslatedName: parentItem.name ??
+                  AppLocalizations.of(context)!.placeholderSource),
+          id: parentItem.id,
+          item: parentItem,
+        ),
+        order: FinampPlaybackOrder.shuffled,
+      );
+    }
+
+    void addArtistNext(List<BaseItemDto> items) {
+      queueService.addNext(
+          items: items,
+          source: QueueItemSource(
+            type: isGenre
+                ? QueueItemSourceType.nextUpPlaylist
```

Comment:
nextUpGenre doesn't exist, is this even particularly important?

Reply:
Not a huge issue. But are you actually adding a whole genre to the queue? 👀

Reply:
I... Didn't think about it that way

Yeah maybe let's not do that... lol